src/lib/MedStickerEncryptionV2.js

```diff
@@ -0,0 +1,85 @@
+import create from "./factory";
+import { arrayBufferToString, encode } from "./utilities";
+import { CHARLIE } from "./constants";
+
+const scrypt = create("scrypt");
+const gcm = create("AES-GCM");
+
+/**
+ *
+ * @param secret
+ * @param salt
+ * @returns {PromiseLike<CryptoKeyPair> | PromiseLike<CryptoKey> | PromiseLike<CryptoKeyPair | CryptoKey>}
+ */
+export async function hash(secret, salt) {
+  try {
+    return await scrypt.generateKey(secret, salt);
+  } catch (e) {
+    throw new Error("HashFailed");
+  }
+}
+
+/**
+ *
+ * @param hashed
+ * @returns {Promise<string>}
+ */
+export function fingerprintSecret(hashed) {
+  const base64EncodedFingerprintSecret = encode(arrayBufferToString(hashed));
+  return `${CHARLIE}-sha256:${base64EncodedFingerprintSecret}`;
+}
+
+/**
+ *
+ * @param array
+ * @returns {{cryptoKey: ArrayBufferLike, accessKey: ArrayBufferLike}}
+ */
+export function splitKeys(array) {
+  const arrayLength = array.length;
+  const cryptoKey = array.slice(0, arrayLength / 2);
+  const fingerprintFile = array.slice(arrayLength / 2, arrayLength);
+
+  return {
+    key: new Uint8Array(cryptoKey).buffer,
+    fingerprintFile: new Uint8Array(fingerprintFile).buffer
+  };
+}
+
+/**
+ *
+ * @returns {string}
+ */
+export function generateRandomAesIv() {
```

Comment:
is that really needed here? I think it could just throw unimplemented error - since you don't need that even in tests

Reply:
> I think it could just throw unimplemented error - since you don't need that even in tests

this will generate the `initializationVector` with some random values and its used in the tests to generate `iv`s so that the `encrypt` function can be fed with it. 
Examples of how its being used in the tests 
[here](https://github.com/VivyTeam/krypt-web/pull/73/files/82ff7b337f23b009024436d4fcbdd67d48388a1e#diff-8c6a8dc4a932988d10415763da2a7dadR36)
and [here](https://github.com/VivyTeam/krypt-web/pull/73/files/82ff7b337f23b009024436d4fcbdd67d48388a1e#diff-8c6a8dc4a932988d10415763da2a7dadR23). 

Probably you mean something i dont get?

Reply:
you don't need random values in tests - just use some predefined bytes.

Thus you can skip this method and limit the scope of the lib

Reply:
yeah but in the real world i would need that function, why shouldnt we have it in one place for all our projects that might be using the lib?

Therefore when we want to update the function with for instance a different implementation we just do it in the lib and not in all our projects?

Reply:
I can ofc delete it no big deal but I think we might wanna have it here.

Reply:
until we gonna activate the sticker in the web (don't think that would happen anytime) - we won't need that method. If we don't need that - why to keep it?

Reply:
ah yeah okay didnt think of that this way. you are right. 

One last thing, there was the idea that the libraries have exactly the same functions exported and the same api for someone to use it, is it now given that, fine, if the other two libs have this function and web doesnt?

Reply:
so you actually already have a different api :) but my idea was to keep the method - just don't implement that, so that signatures will be the same with only difference of the internal details